string agent_id  # TODO: better than an int?
time stamp
ObjectDetection[] objects
